File: src/main/java/io/github/sunshinewzy/shining/api/blueprint/AbstractBlueprintComponent.java
package io.github.sunshinewzy.shining.api.blueprint;

import org.jetbrains.annotations.NotNull;

public abstract class AbstractBlueprintComponent implements IBlueprintComponent {

	private final IBlueprintClass blueprint;

	public AbstractBlueprintComponent(@NotNull IBlueprintClass blueprint) {
		this.blueprint = blueprint;
	}

	@NotNull
	@Override
	public IBlueprintClass getBlueprint() {
		return blueprint;
	}
	
}

File: src/main/kotlin/io/github/sunshinewzy/shining/core/blueprint/AbstractBlueprintComponent.kt
package io.github.sunshinewzy.shining.core.blueprint

import io.github.sunshinewzy.shining.api.blueprint.IBlueprintClass
import io.github.sunshinewzy.shining.api.blueprint.IBlueprintComponent

abstract class AbstractBlueprintComponent(private val blueprint: IBlueprintClass) : IBlueprintComponent {

    override fun getBlueprint(): IBlueprintClass = blueprint
    
}